src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/vsDynamoMesh.hlsl

```diff
@@ -58,9 +57,6 @@ PSInputCustom main(VSInput input)
     //set normal for interpolation	
     output.n = normalize(mul(input.n, (float3x3)mWorld));
 
-    //send our flags through to the frag shader
-    output.customParams.x = vParams.x;
```

Comment:
@mjkkirschner how do the flags get passed to the default `PSInput` struct? We currently have a custom struct for the vertex shader called `VSInputCustom`. Did we also get rid of that in the `helix-upgrade` branch?

Reply:
What do you mean by data interpolation? How did you figure out that that was going on?

Reply:
we no longer need any custom input structs - the data gets passed to each shader using a directX constant buffer, read about these here:
https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-constants

you can think of this like a uniform in opengl except per model, not shader or triangle.



So imagine we have a triangle with 3 verts - each vert has a color associated with it - to smoothly interpolate those colors for each pixel of the triangle, the renderer interpolates those values and sends those interpolated values to the pixel shader - that was being done to the flags data... it turns out that in directX apparently interpolating between the same value on each vertex does not interpolate correctly... the value should remain constant - but it obviously did not... 

in parallels(metal) it did interpolate correctly, so each pixel had the correct flags set.

What we were seeing was each pixel having a random set of flags - some frozen, some selected etc.

I figured out what was going on by noting that the data was being corrupted between the vertex stage and the pixel stage, but was fine if I set it myself using some constant number.

Reply:
Wow, I would have expected interpolation only to be applied to vertex normals and colors, not for all data in the shader.

Reply:
it depends on the semantic you use - check this out:
https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-struct#interpolation-modifiers-introduced-in-shader-model-4